src/tarball.py

```diff
@@ -98,25 +98,24 @@ def _push_tarball(self, config, describe):
         os.unlink(tarball_path)
         return tarball
 
-    def _create_tarball_node(self, checkout_node, status, result):
+    def _create_tarball_node(self, checkout_node, tarball):
         node = {
             'parent': checkout_node['_id'],
             'name': 'tarball',
-            'artifacts': checkout_node['artifacts'],
+            'path': checkout_node['path'] + ['tarball'],
+            'artifacts': {
+                'tarball': urllib.parse.urljoin(self._storage_url, tarball),
+            },
             'revision': checkout_node['revision'],
-            'status': status,
-            'result': result,
+            'status': 'available',
         }
         self._db.submit({'node': node})
 
-    def _update_checkout_node(self, node, describe, version, tarball):
+    def _update_checkout_node(self, node, describe, version):
         node['revision'].update({
             'describe': describe,
             'version': version,
         })
-        node['artifacts'] = {
-            'tarball': urllib.parse.urljoin(self._storage_url, tarball),
-        }
```

Comment:
Why is this removed?

Reply:
The tarball artifact belongs in the tarball node, not the checkout node.